File: 125_valid_palindrome/isPalindrome.py
class Solution(object):
    def isPalindrome(self, s):
        left = 0
        right = len(s) - 1
        s = s.lower()

        while left < right:
            while not s[left].isalnum() and left < right: left += 1
            while not s[right].isalnum() and left < right: right -= 1

            print s[left], s[right]
            if s[left] != s[right]: return False
            else:
                left += 1
                right -= 1
        else:
            return True

if __name__ == '__main__':
    sol = Solution()
    #s = "      A m a#########n, a/////// p   l an, a ca  na  l: Pa   n     a  ma   "
    s = ".....,,,"
    print sol.isPalindrome(s)

File: 9_Palindrome_Number/isPalindrome.cpp
#include <iostream>

using namespace std;

class Solution {
    public:
        bool isPalindrome(int x) {
            /* Negative number is not palindrome. */
            if(x < 0) return false;
            /* x < 10, x must be palindrome. */
            if(x < 10) return true;
            int top = 1, bottom = 10;
            /* Calculate x's number of digits. */
            while(x / top >= 10) top *= 10;
            while(x) {
                /* x / top: left most digit. */
                /* x % bottom: right most digit. */
                if(x / top != x % bottom)
                    return false;
                x %= top;
                x /= bottom;
                /* x loses 2 digits each interation. */
                top /= 100;
            }
            return true;
        }
};

int main() {
    int x = 1;
    Solution sol;
    cout << sol.isPalindrome(x) << endl;

}
